core/src/main/java/io/kestra/core/runners/pebble/functions/RenderFunction.java

```diff
@@ -0,0 +1,58 @@
+package io.kestra.core.runners.pebble.functions;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.runners.VariableRenderer;
+import io.pebbletemplates.pebble.error.PebbleException;
+import io.pebbletemplates.pebble.extension.Function;
+import io.pebbletemplates.pebble.template.EvaluationContext;
+import io.pebbletemplates.pebble.template.EvaluationContextImpl;
+import io.pebbletemplates.pebble.template.PebbleTemplate;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class RenderFunction implements Function {
+    public List<String> getArgumentNames() {
+        return List.of("template", "recursive");
+    }
+
+    @Override
+    public Object execute(Map<String, Object> args, PebbleTemplate self, EvaluationContext context, int lineNumber) {
+        if (!args.containsKey("template")) {
+            throw new PebbleException(null, "The 'render' function expects an argument 'template'.", lineNumber, self.getName());
+        }
+
+        if (!(args.get("template") instanceof String)) {
+            throw new PebbleException(null, "The 'render' function expects an argument 'template' with type string.", lineNumber, self.getName());
+        }
+        String template = (String) args.get("template");
+
+        Object recursiveArg = args.get("recursive");
+        if (recursiveArg == null) {
+            recursiveArg = true;
```

Comment:
I thought we want to discourage recursiveness, in this case shouldn't it be false by default?

Reply:
good catch. By default, we don't render recursively. If you use the render function you opt-in to render (recursively if needed)  explicitly. 

recursive=False is an edge case to render only once as for trigger payload

I'll add a draft PR to docs to show based on examples 👍

Reply:
https://github.com/kestra-io/docs/pull/776